src/main/java/persistence/sql/dml/FindQueryBuilder.java

```diff
@@ -19,26 +21,56 @@ public FindQueryBuilder(Table table) {
     public String build() {
         String columnsClause = buildColumnsClause();
         String tableName = table.getName();
-        return String.format(FIND_ALL_QUERY_FORMAT, columnsClause, tableName);
+
+        String findAllQuery = String.format(Find_QUERY_FORMAT, columnsClause, tableName);
+        String joinClause = buildJoinClause();
+
+        if (joinClause.isEmpty()) {
+            return findAllQuery;
+        }
+        return String.join(" ", findAllQuery, joinClause);
```

Comment:
```suggestion
        String findAllQuery = String.format(Find_QUERY_FORMAT, columnsClause, tableName);
        if (table.hasNoJoinColumns()) {
            return findAllQuery;
        }
        String joinClause = joinQueryBuilder.build();
        return String.join(" ", findAllQuery, joinClause);
```
먼저 joinColumn 을 build 하기 전에 체크할수 있는 방향은 어떠신가요?

Reply:
`FindQueryBuilder`에서 체크하는 것보다 `Table` 객체에 물어보는 것이 더 좋겠네요!
감사합니다